codalab/server/bundle_manager.py

```diff
@@ -327,6 +330,11 @@ def _deduct_worker_resources(self, workers_list):
         """
         for worker in workers_list:
             for uuid in worker['run_uuids']:
+                # If the resource of uuid has been subtracted before, continue to next uuid.
+                # This logic will ensure that the resource of each uuid will only be subtracted once.
+                if uuid in self._track_run_uuids:
+                    continue
+                self._track_run_uuids.add(uuid)
                 bundle = self._model.get_bundle(uuid)
```

Comment:
I think we shouldn't add to the global state here and add a bunch more functions.
Converting the UUID list to a set in the beginning, and then locally skipping failed/killed bundles should be enough like the following:
```suggestion
    worker_run_uuids = set(worker['run_uuids'])
    for uuid in worker_run_uuids:
        bundle = self._model.get_bundle(uuid)
        if bundle.state in (State.KILLED, State.FAILED):
            continue
```